buf/registry/legacy/federation/v1beta1/upload_service.proto

```diff
@@ -89,19 +85,29 @@ message UploadRequest {
     //
     // If the Labels do not exist, they will be created.
     // If the Labels were archived, they will be unarchived.
-    repeated buf.registry.module.v1beta1.ScopedLabelRef scoped_label_refs = 6;
+    repeated buf.registry.module.v1beta1.ScopedLabelRef scoped_label_refs = 5;
     // The URL of the source control commit to associate with the Commit for this Content.
     //
     // BSR users can navigate to this link to find source control information that is relevant to this Commit
     // (e.g. commit description, PR discussion, authors, approvers, etc.).
-    string source_control_url = 7 [
+    string source_control_url = 6 [
       (buf.validate.field).string.uri = true,
       (buf.validate.field).string.max_len = 255,
       (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
     ];
   }
   // The Contents of all references.
   repeated Content contents = 1 [(buf.validate.field).repeated.min_items = 1];
+  // The dependencies of the references specified by Contents.
+  //
+  // This will include all transitive dependencies.
+  //
+  // Dependencies between Contents are implicit and do not need to be specified. The BSR will detect
+  // dependencies between Contents via .proto imports.
+  //
+  // Commits should be unique by Module, that is no two dep_refs should have the same Module but
+  // different Commit IDs.
+  repeated DepRef dep_refs = 2;
```

Comment:
Is it an error to include a DepRef that the BSR prunes away? I.e., if a Commit is referenced here that is never imported (even transitively) by any Content.

Reply:
Not an error I would say